sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -431,6 +431,14 @@ def _receive(
             )
             batch: Union[List["uamqp_Message"], List["pyamqp_Message"]] = []
 
+            while (
+                not received_messages_queue.empty() and len(batch) < max_message_count
+            ):
```

Comment:
why was this moved up here?

Reply:
it was here before, reverting back from last pr